### PR TITLE
BigQuery: Support Stored Procedures

### LINKED ISSUE #3355: Parsing/Templating Error on Stored Procedure
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Sqlfluff is giving templating/parsing error on file which contains the stored procedure. Is stored procedure defined in parsed?. I'm using dialect of bigquery.

### Expected Behaviour

As stored procedure is valid in biquery and runs well, sqlfluff with dialect = bigquery should also parsed it.

### Observed Behaviour

Sqlfluff is giving templating/parsing error on file which contains the stored procedure.

### How to reproduce

Use any valid sql file contain the stored procedure in it,
sqlfluff will throw parsing/templating error on it.

### Dialect

BigQuery

### Version

0.13.1

### Configuration

[sqlfluff]
dialect = bigquery
templater = jinja
exclude_rules = L011,L012,L013,L014,L015,L016,L020,L021,L025,L026,L027,L028,L029,L031,L032,L033,L034,L035,L037,L042,L043,L044,L045,L047,L051,L053,L054,L055,L056,L057,L058,L059,L060,L061,L062

[sqlfluff:rules]
tab_space_size = 2

[sqlfluff:rules:L003]
indent_unit = tab

[sqlfluff:rules:L004]
indent_unit = tab

[sqlfluff:rules:L010]
capitalisation_policy = lower

[sqlfluff:rules:L030]
extended_capitalisation_policy = lower

[sqlfluff:rules:L040]
capitalisation_policy = lower

[sqlfluff:rules:L052]
require_final_semicolon = False

[sqlfluff:rules:L063]
extended_capitalisation_policy = lower

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by tunetheweb**
### Brief summary of the change made

Fixes #3355

### Are there any other side effects of this change that we should be aware of?
Note nested procedural language is not supported (see #3345) so only simple procedures will parse

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_bigquery.py [hunk 1]
```diff
@@ -1622,3 +1651,141 @@ class ExportStatementSegment(BaseSegment):
...
+class ProcedureParameterListSegment(BaseSegment):
+    """The parameters for a prcoedure ie. `(string, number)`."""
+
+    # Procedure parameter list (based on FunctionsParameterListGrammar)
+    type = "procedure_parameter_list"
+    match_grammar = Bracketed(
+        Delimited(
+            Ref("ProcedureParameterGrammar"),
+            delimiter=Ref("CommaSegment"),
+            bracket_pairs_set="angle_bracket_pairs",
```
[the remark this answers is not in the document]
Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_bigquery.py:1686]: This is copied from `FunctionParameterListGrammar` in BigQuery and it's specifically listed there. I think there's a good reason but can't quite remember what it is. Something about parameters in this section having to use angle brackets for STRUCT type definition.

So something like this:

```py
CREATE PROCEDURE example (param1 INT, param2 STRUCT<x INT64>)
```

To be honest I suspect some of this is part of that other bug where it saw the wrong bracket types, and so wrong grammar without `parse_grammar` being defined (even though it was same as `match_grammar).

I agree you shouldn't have to redefine the bracket types in here, but for some reason you do.

[the remark this answers is not in the document]
Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_bigquery.py:1686]: Ah ha, found the original implementation: https://github.com/sqlfluff/sqlfluff/pull/997

So this is basically a repeat of this. As I say, a straight copy from `FunctionParameterListGrammar`

[the remark this answers is not in the document]
Comment (WittierDinosaur) [reply src/sqlfluff/dialects/dialect_bigquery.py:1686]: Fair enough - just making sure 👍

**Review by WittierDinosaur — APPROVED** [review]
LGTM

**Pull request merged**